celery_tasks_prioritizing/README.md

```diff
@@ -0,0 +1,105 @@
+# Prioritizing of the Celery tasks
+
+## Current state
+
+- 2 workers in prod, 1 worker in staging, no priority
+- in each worker we do:
+
+`celery --app="${APP}" worker --loglevel=${LOGLEVEL} --concurrency=1 --prefetch-multiplier=1`
+
+## What do we want to achieve
+
+- process the tasks depending on their priority and duration
+
+- high-prio:
+  - actionable events to users such as results with logs, final events
+  - informations with high value - i.e. first status report
+- low-prio:
+  - intermediate results
+
+| high-prio                      | low-prio                         |
+| ------------------------------ | -------------------------------- |
+| process_message                | run_copr_build_start_handler     |
+| run_copr_build_handler         | run_copr_build_end_handler       |
+| run_testing_farm_handler       | run_koji_build_report_handler    |
+| run_propose_downstream_handler | run_testing_farm_results_handler |
+| run_koji_build_handler         | run_installation_handler         |
+| run_pagure_pr_label_handler    | babysit_copr_build               |
+|                                | run_distgit_commit_handler       |
+
+- long-running
+  - tasks that build the SRPM: run_koji_build_handler, run_copr_build_handler
+- short-running:
+  - other tasks
```

Comment:
From Martin Pitt:

We do that in Cockpit as well. We have a "webhook" AMQP queue which always gets drained first -- these are high-priority fast tasks such as "react to this pull_request webhook event, interpret it, and put that into the tasks queue", and a "tasks" queue with the slow and bulky stuff ("exercise a test", "build a
VM image").